sycl/test-e2e/Matrix/joint_matrix_prefetch_impl.hpp

```diff
@@ -0,0 +1,141 @@
+  auto pB = address_space_cast<sycl::access::address_space::global_space,
+                               access::decorated::no>(B);
+  auto pC = address_space_cast<sycl::access::address_space::global_space,
+                               access::decorated::no>(C);
+
+  joint_matrix<sub_group, T1, use::a, TM, TK, layout::row_major> sub_a;
+  joint_matrix<sub_group, T2, use::b, TK, TN, layout::row_major> sub_b;
+  joint_matrix<sub_group, T, use::accumulator, TM, TN> sub_c;
+  joint_matrix_prefetch(sg, A + (sg_startx * TM) * K, K, layout::row_major, TM,
+                        TK, syclex::properties{syclex::prefetch_hint_L1});
+  joint_matrix_prefetch(sg, B + sg_starty / SG_SZ * TN, N, layout::row_major,
+                        TK, TN, syclex::properties{syclex::prefetch_hint_L2});
+  joint_matrix_prefetch(sg, C + (sg_startx * TM) * N + sg_starty / SG_SZ * TN,
+                        N, layout::row_major, TM, TN,
+                        syclex::properties{syclex::prefetch_hint_L1});
+  joint_matrix_fill(sg, sub_c, 1);
+  for (int k = 0; k < K; k += TK) {
+    joint_matrix_load(sg, sub_a, pA + (sg_startx * TM) * K + k, K);
+    joint_matrix_load(sg, sub_b, pB + k * N + sg_starty / SG_SZ * TN, N);
+    joint_matrix_mad(sg, sub_c, sub_a, sub_b, sub_c);
+  }
+  joint_matrix_store(sg, sub_c,
+                     pC + (sg_startx * TM) * N + sg_starty / SG_SZ * TN, N,
+                     layout::row_major);
+}
+
+template <typename T, typename T1, typename T2, size_t M, size_t N, size_t K,
+          unsigned int vnniFactor>
+void matrix_multiply(T *C, T1 *A, T2 *B, queue q) {
+  size_t NDRangeM = M / TM;
+  size_t NDRangeN = N / TN;
+
+  q.submit([&](handler &cgh) {
+     cgh.parallel_for(
+         nd_range<2>({NDRangeM, NDRangeN * SG_SZ}, {1, 1 * SG_SZ}),
+         [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(SG_SZ)]]
+
+         {
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           sub_group sg = spmd_item.get_sub_group();
+           if constexpr (vnniFactor > 1)
+             joint_matrix_gemm_vnni<T1, T2, T, M, N, K, vnniFactor>(
+                 sg, sg_startx, sg_starty, A, B, C);
+           if constexpr (vnniFactor == 1)
+             joint_matrix_gemm<T1, T2, T, M, N, K>(sg, sg_startx, sg_starty, A,
+                                                   B, C);
+         }); // parallel for
+   }).wait();
+}
+
+int main() {
+  queue q;
+  std::vector<combination> combinations =
+      q.get_device()
+          .get_info<sycl::ext::oneapi::experimental::info::device::
+                        matrix_combinations>();
+  bool support_p = false;
+  for (int i = 0; i < combinations.size(); i++) {
+    if (combinations[i].atype == matrix_type::tf32)
+      support_p = true;
+  }
+  if (support_p) {
```

Comment:
We can reduce indentation level by inversing this condition:
```suggestion
  if (!support_p)
      return 0;
```